visual-java/src/main/java/com/saucelabs/visual/VisualApi.java

```diff
@@ -283,6 +322,41 @@ public VisualApi(
     this.sessionMetadataBlob = sessionMetadataBlob;
   }
 
+  /**
+   * Use reflection to attempt to automatically determine if the current session is Sauce if not
+   * already configured by the user or cached.
+   */
+  private boolean isSauceSession() {
+    if (isSauceSession != null) {
+      return isSauceSession;
+    }
+
+    CommandExecutor executor = this.driver.getCommandExecutor();
+    Object resolvedExecutor = null;
+    if (executor instanceof TracedCommandExecutor) {
+      try {
+        final Field field = TracedCommandExecutor.class.getDeclaredField("delegate");
+        field.setAccessible(true);
+        resolvedExecutor = field.get(executor);
+      } catch (NoSuchFieldException | IllegalAccessException ignored) {
```

Comment:
What do you think about logging these? If something goes wrong, it will be difficult to debug

Reply:
I _think_ we'd be fine without it? We wouldn't get client side logs anyway, but this is just to eager set that value for the defaults in Selenium. it will default to Sauce if we can't verify it through this method, and users that want to force local can use our configuration options to skip this automatic detection.